Docs: add inputs and outputs in workflow (#266)
oam-dev/kubevela.io: docs/end-user/component-dependency-parameter.md

This section will introduce how to pass data between components.

## Inputs and Outputs

In KubeVela, we can use inputs and outputs in Components to pass data.

### Outputs

```go
type Outputs []outputItem

type outputItem struct {
	ValueFrom string `json:"valueFrom"`
	Name      string `json:"name"`

> }

Reviewer: 最好别写go结构体给最终用户，一看就头大

```

Outputs is made of `name` and `valueFrom`. Input will use `name` to reference output.

We can write `valueFrom` in the following ways:

> 1. Use string, eg. `valueFrom: testString`

Reviewer: ```suggestion
1. Fill string value in the field, eg. `valueFrom: testString`
```

> 2. Use expression, eg. `valueFrom: output.metadata.name`. Note that `output` is a built-in field referring to the resource in the component that is deployed to the cluster.

Reviewer: ```suggestion
2. Use expression, eg. `valueFrom: output.metadata.name`. Note that `output` is a built-in field referring to the resource in the component that is rendered and deployed to the cluster.
```

> 3. Use `+` to combine above two ways, eg. `valueFrom: output.metadata.name + "testString"`

Reviewer: ```suggestion
3. Use `+` to combine above two ways, the computed value will be the result, eg. `valueFrom: output.metadata.name + "testString"`
```

### Inputs

```go
type Inputs []inputItem

type inputItem struct {
	ParameterKey string `json:"parameterKey"`
	Name         string `json:"name"`
}
```

Inputs is made of `name` and `parameterKey`. Input uses `name` to reference output, `parameterKey` is a expression that assigns the value of the input to the corresponding field.

eg.

1. Specify inputs:

```yaml
...
- name: wordpress
  type: helm
  inputs:

>     - from: mysql-svc

Reviewer: ```suggestion
    - from: mysql-svc
       parameterKey: properties.values.externalDatabase.host    
```

```

> 2. The field `properties.values.externalDatabase.host` in component will be assigned after rendering:

Reviewer: ```suggestion
2. The field parameterKey specifies the field path of the parameter key in component to be assigned after rendering:
```

> ```yaml
> ...
> - name: wordpress
>   type: helm
>   inputs:
>     - from: mysql-svc
>       parameterKey: properties.values.externalDatabase.host
>   properties:
>     values:
>       externalDatabase:
>         host: "value"
> ```

Reviewer: ```suggestion
Which means the input value will be passed into the belowing properties:
```yaml
...
- name: wordpress
  type: helm
  properties:
    values:
      externalDatabase:
        host: <input value>
```
```
